### DISCUSSION TITLE
.log files?

### DISCUSSION THREAD

**Discussion by sannatitus** [bonsai-rx/bonsai#1162] [Q&A]
Hi everyone 🌳

Does Bonsai produce .log files, and if so, where do I find them? 

Alternatively, does anyone know what the attached Reliability Monitor error means for Bonsai? 

I’ve been experiencing some crashing of Bonsai on my desktop acquiring video from two Allied Vision cameras. Bonsai crashes coincide with Windows hardware errors. I believe the issue is that I need to update my graphics card driver (ie, the Windows error attached), but I would like to confirm this. 

Thanks for your help everyone,
Sanna 

![2F76B61D-0DF0-41AE-849E-C051359237D1](https://user-images.githubusercontent.com/100142813/204770014-ddbc06b8-ade8-4666-a787-99f0533b92e3.jpeg)
![4F5FCC42-59A4-4EF8-B649-5ED844DD24FE](https://user-images.githubusercontent.com/100142813/204770057-182f6bef-bbcc-4e51-a3a6-8e37175b26d3.jpeg)
![D27D6C0C-2ABB-4D66-B721-6CD1E9526AAE](https://user-images.githubusercontent.com/100142813/204770104-3e9dd057-0260-4999-8437-d4babeb9870e.jpeg)

**Comment by bruno-f-cruz** [comment]
You can try to pipe the output of the [`stderr`](https://en.wikipedia.org/wiki/Standard_streams) to a text file via the command line. For simplicity, I usually just use a python script that takes care of it for me :P 
Something like this:

```python

import subprocess
import os
from pathlib import Path

ADD_FLAGS = "--no-editor" 
SAVE_LOG = True
log_dir = "logdir"
bonsai_path = r"pathtoexe\Bonsai.exe"
workflow_path = r"pathtoworkflow\myworkflow.bonsai"
cwd = Path(__file__).parent.resolve()

output_cmd = f'"{bonsai_path}" "{workflow_path}" {ADD_FLAGS}'

if SAVE_LOG:
    os.mkdir(log_dir)
    log = open(log_dir + "\\" + 'log.txt', 'a')  
    bonsai_process = subprocess.Popen(output_cmd, cwd=cwd, creationflags=subprocess.CREATE_NEW_CONSOLE, stdout=log, stderr=log)
else:
    bonsai_process = subprocess.Popen(output_cmd, cwd=cwd, creationflags=subprocess.CREATE_NEW_CONSOLE)

```
